### ISSUE TITLE
ValidationError override multiple errors on the same field

### ISSUE THREAD

**Issue by ningirsu** [recurly/recurly-client-python#197]
Hi!

When you have a validation error with multiple errors on the same field, only the last message is kept.

Eg. On a subscription error you can have a xml like that:
```xml

<errors>
  <error field="subscription.account.billing_info.address1" symbol="empty">can't be empty</error>
  <error field="subscription.account.billing_info.city" symbol="empty">can't be empty</error>
  <error field="subscription.account.billing_info.country" symbol="empty">can't be empty</error>
  <error field="subscription.account.billing_info.first_name" symbol="blank">can't be blank</error>
  <error field="subscription.account.billing_info.last_name" symbol="blank">can't be blank</error>
  <error field="subscription.account.billing_info.number" symbol="required">is required</error>
  <error field="subscription.account.billing_info.zip" symbol="empty">can't be empty</error>
  <error field="subscription.account.account_code" symbol="blank">can't be blank</error>
  <error field="subscription.account.account_code" symbol="invalid">is invalid</error>
  <error field="subscription.plan_code" symbol="invalid">is invalid</error>
  <error field="subscription.unit_amount_in_cents" symbol="not_a_number">is not a number</error>
</errors>
```
And the python representation of the error is:
```python

In [98]: try:
    ...:     recurly.Subscription(account=recurly.Account(billing_info=recurly.BillingInfo())).save()
    ...: except recurly.ValidationError as err:
    ...:     pass
    ...:     

In [99]: print(err)
not_a_number: subscription.unit_amount_in_cents is not a number; invalid: subscription.plan_code is invalid; required: subscription.account.billing_info.number is required; empty: subscription.account.billing_info.address1 can't be empty; blank: subscription.account.billing_info.last_name can't be blank; blank: subscription.account.billing_info.first_name can't be blank; empty: subscription.account.billing_info.country can't be empty; empty: subscription.account.billing_info.zip can't be empty; empty: subscription.account.billing_info.city can't be empty; invalid: subscription.account.account_code is invalid

In [100]: err.errors
Out[100]: 
{'subscription.account.account_code': <recurly.errors.Suberror at 0x7f0b0ece8c90>,
 'subscription.account.billing_info.address1': <recurly.errors.Suberror at 0x7f0b1c161cd0>,
 'subscription.account.billing_info.city': <recurly.errors.Suberror at 0x7f0b1c161590>,
 'subscription.account.billing_info.country': <recurly.errors.Suberror at 0x7f0b1c161850>,
 'subscription.account.billing_info.first_name': <recurly.errors.Suberror at 0x7f0b1c161390>,
 'subscription.account.billing_info.last_name': <recurly.errors.Suberror at 0x7f0b1c161190>,
 'subscription.account.billing_info.number': <recurly.errors.Suberror at 0x7f0b0ece8290>,
 'subscription.account.billing_info.zip': <recurly.errors.Suberror at 0x7f0b0ece8dd0>,
 'subscription.plan_code': <recurly.errors.Suberror at 0x7f0b0ece8810>,
 'subscription.unit_amount_in_cents': <recurly.errors.Suberror at 0x7f0b0ece8fd0>}

```
We see that we only have one ```subscription.account.account_code``` instead of two

Could we have a list of suberrors as a value of the dict instead of just one suberror ?

The code is located here: https://github.com/recurly/recurly-client-python/blob/master/recurly/errors.py#L238
```python
suberrors = dict()
for err in self.response_doc.findall('error'):
    field = err.attrib['field']
    symbol = err.attrib['symbol']
    message = err.text

    suberrors[field] = self.Suberror(field, symbol, message)

self.__dict__['errors'] = suberrors
```
Maybe use a defaultdict(list) instead of a dict ?

**Comment by bhelx** [comment]
@ningirsu Thanks a lot for the highly detailed report! At first glance, your suggestion of using `defaultdict(list)` makes sense. I think that was probably the original intention. I'll get a fix prepared.

**Comment by bhelx** [comment]
Sorry this took so long to address. @ningirsu perhaps you can give your input on the solution: https://github.com/recurly/recurly-client-python/pull/208

**Comment by pakal** [comment]
Erh nope the #208 was merged but it brings more problems than it solves. 

Not only it breaks retrocompatibility with existing users of the package, who don't expect a list were there were only simple SubErrors, but it breaks other internal tools like stringification (repr() ends up being used): 

```
    def __unicode__(self):
        return six.u('; ').join(six.text_type(error)
for error in six.itervalues(self.errors))
```

Keeping retrocompat' would maybe involve adding arguments like "as_list (default False)", in such cases.

**Comment by pakal** [comment]
What is the plan for validationError.__str__ ? Something like this ?

```
def validationerror__unicode__(self):
    all_error_strings = []
    for error in self.errors.values():
        if isinstance(error, (tuple, list)):
            all_error_strings += [six.text_type(e) for e in error]  # multiple errors on field
        else:
            all_error_strings.append(six.text_type(error))
    return six.u('; ').join(all_error_strings)
```

**Comment by bhelx** [comment]
@pakal can we move this conversation back to #217?

This change was not meant to be backwards compatible, that's why it was released on `2.6.0` with upgrade notes rather than `2.5.1`.

**Comment by pakal** [comment]
Ok seen  B-)

**Comment by aaron-suarez** [comment]
This appears to be fixed by #208 and #218. Closing out this issue
